lib/turn.rb

```diff
@@ -0,0 +1,61 @@
+require 'pry'
+require_relative './card'
+require_relative './deck'
+require_relative './player'
+
+
+class Turn
+    attr_reader :player1, :player2, :spoils_of_war
+
+    def initialize(player1, player2)
+        @player1 = player1
+        @player2 = player2
+        @spoils_of_war = []
+    end
+
+    def type
+        if @player1.deck.rank_of_card_at(0) != @player2.deck.rank_of_card_at(0)
+            return :basic
+        elsif @player1.deck.cards[2] == nil || @player2.deck.cards[2] == nil
```

Comment:
I like this extra condition that checks whether a player even has enough cards in their deck and then performing some alternative action based on this condition.